src/features/auth/authSaga.js

```diff
@@ -1,50 +1,126 @@
-import { signInWithPopup, GoogleAuthProvider } from "firebase/auth";
+import {
+  signupRequest,
+  loginRequest,
+  loginSuccess,
+  loginFailure,
+  logoutRequest,
+  logoutSuccess,
+  logoutFailure,
+} from "./authSlice";
+import userApi from "../../utils/api/user";
 
-// 예시 추후엔 별도의 모듈로 관리...
-const loginApi = ({ token }) => {
-  axios.post("/login", { token: token }, { withCredentials: true });
-};
+function* userLogin({ payload }) {
+  try {
+    if (payload) {
+      const { email, password } = payload;
+      const response = yield signInWithEmailAndPassword(
+        authenication,
+        email,
+        password
+      );
+      const { accessToken: token } = response.user;
 
-function* userLogin() {
-  const provider = new GoogleAuthProvider();
+      const apiResult = yield call(userApi.getlogin, token);
 
-  try {
-    const response = yield signInWithPopup("/*인증*/", provider);
-    const { accessToken: token } = response.user;
+      if (apiResult.result === "success") {
+        yield put(
+          loginSuccess({
+            email: response.user.email,
+            name: response.user.displayName,
+          })
+        );
+      } else {
+        yield put(loginFailure(apiResult.error));
+      }
+    } else {
+      const provider = new GoogleAuthProvider();
+      const response = yield signInWithPopup(authenication, provider);
+      const { accessToken: token } = response.user;
+
+      const apiResult = yield call(userApi.getlogin, token);
+
+      if (apiResult.result === "success") {
+        yield put(
+          loginSuccess({
+            email: response.user.email,
+            name: response.user.displayName,
+          })
+        );
+      } else {
+        yield put(loginFailure(apiResult.error));
+      }
+    }
+  } catch (err) {
+    yield put(loginFailure(err));
+  }
+}
 
-    // const { result, user, error }  = yield axios.post("/login", { token: token }, { withCredentials: true });
+function* userSignup(action) {
+  const { email, name, password } = action.payload;
 
-    const { result, user } = yield call(loginApi, token);
+  try {
+    const response = yield createUserWithEmailAndPassword(
+      authenication,
+      email,
+      password
+    );
+    yield updateProfile(authenication.currentUser, { displayName: name });
 
-    if (result === "ok") {
-      yield put(login({ email: user.email, name: user.name }));
+    const { accessToken: token } = response.user;
+    const apiResult = yield call(userApi.getlogin, token);
+
+    if (apiResult.result === "success") {
+      yield put(
+        loginSuccess({
+          email,
+          name,
+        })
+      );
+    } else {
+      yield put(loginFailure(apiResult.error));
     }
   } catch (err) {
-    // 에러 처리
+    yield put(loginFailure(err));
   }
 }
 
-function* userSginup() {}
+function* userLogout() {
+  try {
+    yield signOut(authenication);
 
-// react 컴포너트
-// dipatch("USER_LOGIN")
+    const apiResult = yield call(userApi.getlogout);
```

Comment:
```suggestion
    const response = yield call(userApi.getlogout);
```
response가 조금 더 나은것 같아요 아래 코드에서 apiResult.result 이라서 조금 어색해 보이는 것 같아요

Reply:
네! 이부분도 위에 수인님께서 남겨주신 코멘트처럼 response를 어디에서 받느냐에 따라 변수명 수정하도록 하겠습니다.